internal/client/consensus/common/block_import.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2025 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package common
+
+import (
+	"github.com/ChainSafe/gossamer/internal/primitives/consensus/common"
+	"github.com/ChainSafe/gossamer/internal/primitives/runtime"
+	statemachine "github.com/ChainSafe/gossamer/internal/primitives/state-machine"
+	"github.com/ChainSafe/gossamer/internal/primitives/state-machine/overlayedchanges"
+)
+
+type ImportResult interface {
+	isImportResult()
+}
+
+// Auxiliary data associated with an imported block result.
+type importedAux struct {
+	// Only the header has been imported. Block body verification was skipped.
+	HeaderOnly bool
+	// Clear all pending justification requests.
+	ClearJustificationRequests bool
+	// Request a justification for the given block.
+	NeedsJustification bool
+	// Received a bad justification.
+	BadJustification bool
+	// Whether the block that was imported is the new best block.
+	IsNewBest bool
+}
+
+type (
+	// Block imported.
+	Imported struct{ importedAux }
+	// Already in the blockchain.
+	AlreadyInChain struct{}
+	// Block or parent is known to be bad.
+	KnownBad struct{}
+	// Block parent is not in the chain.
+	UnknownParent struct{}
+	// Parent state is missing.
+	MissingState struct{}
```

Comment:
I usually prefix it with `ImportResult` to make it clear.
```suggestion
	ImportResultMissingState struct{}
```